bundle-core/src/main/java/net/discdd/bundlesecurity/ServerSecurity.java

```diff
@@ -192,7 +193,14 @@ private void updateSessionRecord(ClientSession clientSession) {
 
     private void initializeClientKeysFromFiles(Path path, ClientSession clientSession) throws IOException,
             InvalidKeyException {
-        byte[] clientIdentityKey = SecurityUtils.decodePublicKeyfromFile(path.resolve(CLIENT_IDENTITY_KEY));
+        byte[] clientIdentityKey;
+        try {
+            String clientIdentityKeyBase64 =
+                    SecurityUtils.decodeEncryptedPublicKeyfromFile(ourIdentityKeyPair.getPrivateKey(), path.resolve(CLIENT_IDENTITY_KEY));
+            clientIdentityKey = Base64.getUrlDecoder().decode(clientIdentityKeyBase64);
+        } catch (NoSuchAlgorithmException e) {
+            throw new RuntimeException(e);
```

Comment:
i think this should be an InvalidKeyException: new InvalidKeyException("No such algorithm", e);